In [1]:
%load_ext autoreload
%autoreload 2

In [2]:
%pip install numpy
%pip install statistics
%pip install scipy
%pip install matplotlib
%pip install pandas
%pip install pytest

Defaulting to user installation because normal site-packages is not writeable

[notice] A new release of pip is available: 26.0.1 -> 26.2.1
[notice] To update, run: python3 -m pip install --upgrade pip
Note: you may need to restart the kernel to use updated packages.
Defaulting to user installation because normal site-packages is not writeable

[notice] A new release of pip is available: 26.0.1 -> 26.2.1
[notice] To update, run: python3 -m pip install --upgrade pip
Note: you may need to restart the kernel to use updated packages.
Defaulting to user installation because normal site-packages is not writeable

[notice] A new release of pip is available: 26.0.1 -> 26.2.1
[notice] To update, run: python3 -m pip install --upgrade pip
Note: you may need to restart the kernel to use updated packages.
Defaulting to user installation because normal site-packages is not writeable

[notice] A new release of pip is available: 26.0.1 -> 26.2.1
[notice] To update, run: python3 -m pip install --upgrad

### Solar Micro-Grid Dispatch Planner

In [3]:
from src.microgrid import MicroGrid

micro_grid = MicroGrid()
determinant = micro_grid.determinant()
condition = micro_grid.condition()

print(f"Determinant is: {determinant}", f"Condition: {condition}")

Determinant is: -4.999999999999999 Condition: 5.82842712474619


### Explain determinant and condition
**Determinant being -4.99999 or 5.** This means the co-efficients are meaningful and this usable to solve the system, if it was 0, it would be the system's equation are not solveable  
**Condtion being 5.8284...** A condition number of about 5.8284 means a 1% error in D1 or D2 can cause at most about a 5.8284% error in the values of x or y. That's small, so the system is well-conditioned. A large condition number , say more that 100, in 1000s or more, would cause significant measurement errors

### Interactive Input

In [24]:
from src.validate import Validate

try:
    validate = Validate()
    
    d1_interactive = input('Provide daytime load (KWh): ')
    validate.check_energy_input(d1_interactive)

    d2_interactive = input('Provide critical-equipment load (kWh):')
    validate.check_energy_input(d2_interactive)
    
    #solve
    result = micro_grid.solve(float(d1_interactive),float(d2_interactive))
    
    print("Results: ",result)
except Exception as e:
    print(e)    



Results:  [ 1.6 -0.4]


### CSV input


In [ ]:
from src.csv_generator import CsvGenerator
from pathlib import Path
import csv

csv_file_path = './files/solar_data.csv'
csv_generator = CsvGenerator()

#generate file
csv_generator.solar_grid_csv(csv_file_path,30,36)

#check if csv file exists
if Path(csv_file_path).exists():
    print(f"File {csv_file_path} generate successfully!")


Day 1: [11.39 31.2 ]
Day 2: [ 8.508 41.828]
Day 3: [ 8.94 39.49]
Day 4: [10.71 28.94]
Day 5: [10.302 32.022]
Day 6: [ 4.606 32.416]
Day 7: [ 4.256 34.046]
Day 8: [ 9.386 39.196]
Day 9: [10.334 32.954]
Day 10: [ 5.838 45.138]
Day 11: [ 9.49 32.  ]
Day 12: [ 7.078 42.028]
Day 13: [ 6.414 27.224]
Day 14: [ 6.866 27.946]
Day 15: [ 9.85 31.91]
Day 16: [11.174 32.094]
Day 17: [11.574 29.024]
Day 18: [ 7.614 37.754]
Day 19: [ 9.136 33.376]
Day 20: [ 5.078 36.678]
Day 21: [ 2.866 40.096]
Day 22: [ 9.096 33.716]
Day 23: [ 6.258 42.108]
Day 24: [ 3.724 49.284]
Day 25: [ 7.4  36.25]
Day 26: [ 6.886 36.346]
Day 27: [ 4.334 30.894]
Day 28: [ 2.494 40.654]
Day 29: [ 6.648 41.278]
Day 30: [ 9.676 32.186]
